virtool/auth/client.py

```diff
@@ -0,0 +1,126 @@
+import asyncio
+from abc import ABC, abstractmethod
+from typing import Union
+
+from openfga_sdk import ApiClient, TupleKey
+from virtool_core.models.enums import Permission
+
+from virtool.api.response import NotFound
+from virtool.auth.mongo import (
+    check_in_mongo,
+    list_permissions_in_mongo,
+)
+from virtool.auth.openfga import (
+    check_in_open_fga,
+    list_permissions_in_open_fga,
+    add_in_open_fga,
+    remove_in_open_fga,
+)
+from virtool.auth.relationships import BaseRelationship
+from virtool.data.layer import DataLayer
+
+from virtool.mongo.core import DB
+
+
+class AbstractAuthorizationClient(ABC):
+    @abstractmethod
+    async def check(
+        self,
+        user_id: str,
+        permission: Permission,
+        object_type: str,
+        object_id: Union[str, int],
+    ) -> bool:
+        ...
+
+    @abstractmethod
+    async def list_permissions(
+        self, user_id: str, object_type: str, object_id: Union[str, int]
+    ) -> dict:
```

Comment:
This should return a `list`.

Reply:
Would this just be a list of the allowed permissions without the boolean value? Right now I am returning a dict of permissions such as: 
{'cancel_job': True, 'create_ref': False, 'create_sample': False, 'modify_hmm': False, 'modify_subtraction': False, 'remove_file': False, 'remove_job': False, 'upload_file': False}.

Reply:
We'll figure this out on Monday.

I want to move away from the dictionary model. It doesn't serve much purpose and will be a burden once permissions start spreading to resources other than `app`.

The problem is that the API returns permissions objects like the one shown in your comment. We'll have to strategize on this.